### PR TITLE
Improve test coverage of video playback

### PR DESCRIPTION
**Pull request by Opelkuh**
- [x] Depends on #4892 for one of the .webm tests to pass

Adds a simple video that I made to resources in 3 different formats, .mp4 and .avi with H.264 and .webm with VP9. I tried to make them as small as possible but they still take ~1MB.
Preview of the beauty:
https://raw.githubusercontent.com/Opelkuh/osu-framework/25ba3e0a2b1041f42eb9fb91d7509326587d8c1c/osu.Framework.Tests/Resources/Videos/sample-video.mp4

I made sure that it would've caught the .avi issues from #4858

### CODE REVIEW

**Review by bdach — CHANGES_REQUESTED** [review]

File: osu.Framework.Tests/Visual/Sprites/TestSceneVideo.cs [hunk 1]
```diff
@@ -45,33 +49,31 @@ private void load()
...
-            AddStep("Reset clock", () => clock.CurrentTime = 0);
-        }
-
-        private void loadNewVideo()
-        {
-            AddStep("load video", () =>
+            AddStep($"load .{format} video", () =>
             {
-                videoContainer.Child = video = new TestVideo
+                videoContainer.Child = video = new TestVideo(videoStore.GetStream($"sample-video.{format}"))
```
[the remark this answers is not in the document]
Comment (Opelkuh) [reply osu.Framework.Tests/Visual/Sprites/TestSceneVideo.cs:61]: I think it gets disposed by `VideoDecoder`: https://github.com/Opelkuh/osu-framework/blob/25ba3e0a2b1041f42eb9fb91d7509326587d8c1c/osu.Framework/Graphics/Video/VideoDecoder.cs#L856

File: osu.Framework.Tests/Visual/FrameworkGridTestScene.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Testing;
+
+namespace osu.Framework.Tests.Visual
+{
+    public abstract class FrameworkGridTestScene : GridTestScene
```
[the remark this answers is not in the document]
Comment (Opelkuh) [reply osu.Framework.Tests/Visual/FrameworkGridTestScene.cs:8]: Yes. It's only to get the `DllResourceStore` loaded for `TestSceneVideoLayout` and `TestSceneWrapModes`.

File: osu.Framework.Tests/Visual/Sprites/TestSceneVideo.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (bdach) [reply osu.Framework.Tests/Visual/Sprites/TestSceneVideo.cs:61]: Oh right, my bad. It's arguably a little hidden, but I should have spotted that either way.

File: osu.Framework.Tests/Visual/Sprites/TestSceneVideoLayout.cs [hunk 3]
```diff
@@ -2,39 +2,44 @@
...
 
         [BackgroundDependencyLoader]
-        private void load()
+        private void load(Game game)
         {
-            Cell(0, 0).Child = createTest("video - auto size", () => new TestVideo());
-            Cell(0, 1).Child = createTest("video - relative size + fit", () => new TestVideo
+            byte[] videoData = game.Resources.Get("Videos/sample-video.mp4");
+
+            Cell(0, 0).Child = createTest("video - auto size", () => new TestVideo(new MemoryStream(videoData)));
```
Comment (peppy) [inline osu.Framework.Tests/Visual/Sprites/TestSceneVideoLayout.cs:28]: Is there a reason these can't be direct calls to `game.Resources.GetStream`?

Comment (Opelkuh) [reply osu.Framework.Tests/Visual/Sprites/TestSceneVideoLayout.cs]: Nope.

**Review by bdach — APPROVED** [review]
looks fine here, but going to wait for a double check

**Review by peppy — APPROVED** [review]

**Pull request merged**